lib/vagrant-service-manager/plugin_util.rb

```diff
@@ -17,6 +17,37 @@ def self.host_docker_path(machine)
         # Path to the private_key and where we will store the TLS Certificates
         File.expand_path("docker", machine.data_dir)
       end
+
+      def self.machine_ip(machine)
+        machine.guest.capability(:machine_ip)
+      end
+
+      def self.sha_id(file_data)
+        Digest::SHA256.hexdigest file_data
+      end
+
+      def self.certs_present_and_valid?(path, machine)
+        return false if Dir["#{path}/*"].empty?
+
+        # check validity of certs
+        Dir[path + "/*"].each do |f|
+          guest_file_path = "#{DOCKER_PATH}/#{File.basename(f)}"
+          guest_sha = machine.guest.capability(:sha_id, guest_file_path)
+          return false if sha_id(File.read(f)) != guest_sha
+        end
+
+        true
+      end
+
+      def self.service_running?(machine, service)
```

Comment:
@bexelbie That make sense to me :smile: 
This way, each service will able to determine whether they are running or not by themselves.